modules/python-modules/syslogng/modules/webhook/source.py

```diff
@@ -52,6 +66,18 @@ async def post(self, **path_arguments) -> None:
 
         await self.finish({"status": "received"})
 
+    def _set_proxied_ip(self, msg: LogMessage) -> None:
+        proxy_header_candidates = [self.request.headers.get_list(header) for header in WEBHOOK_PROXY_SOURCEIP_HEADERS]
+
+        for header in proxy_header_candidates:
+            if header and len(header) > 0:
+                # the closest/last IP (the behind_proxy flag implies that the last one can be trusted)
+                msg.set_source_ipaddress(header[-1])
+                msg[WEBHOOK_NV_PREFIX + "proxy_ip"] = self.request.remote_ip
```

Comment:
in the alternate branch for a similar case, we were using $PEER_ADDR to store this information, which should be handled similarly here.

Either using a 3rd GSockAddr in LogMessage or a name-value pair, but one that matches between webhook and syslog-style sources.

Reply:
I think everything has become consistent.